--- a/src/main/include/log4cxx/helpers/simpledateformat.h
+++ b/src/main/include/log4cxx/helpers/simpledateformat.h
@@ -27,10 +27,9 @@

 #include <log4cxx/helpers/dateformat.h>
 #include <vector>
+#include <locale>
 #include <time.h>

-namespace std { class locale; }
-
 namespace log4cxx
 {
         namespace helpers
--- a/src/main/cpp/stringhelper.cpp
+++ b/src/main/cpp/stringhelper.cpp
@@ -28,6 +28,7 @@
 #endif
 #include <log4cxx/private/log4cxx_private.h>
 #include <cctype>
+#include <cstdlib>
 #include <apr.h>


--- a/src/main/cpp/inputstreamreader.cpp
+++ b/src/main/cpp/inputstreamreader.cpp
@@ -20,7 +20,9 @@
 #include <log4cxx/helpers/exception.h>
 #include <log4cxx/helpers/pool.h>
 #include <log4cxx/helpers/bytebuffer.h>
+#include <cstring>

+using namespace std;
 using namespace log4cxx;
 using namespace log4cxx::helpers;

--- a/src/main/cpp/socketoutputstream.cpp
+++ b/src/main/cpp/socketoutputstream.cpp
@@ -19,7 +19,9 @@
 #include <log4cxx/helpers/socketoutputstream.h>
 #include <log4cxx/helpers/socket.h>
 #include <log4cxx/helpers/bytebuffer.h>
+#include <cstring>

+using namespace std;
 using namespace log4cxx;
 using namespace log4cxx::helpers;

--- a/src/examples/cpp/console.cpp
+++ b/src/examples/cpp/console.cpp
@@ -22,7 +22,10 @@
 #include <log4cxx/logmanager.h>
 #include <iostream>
 #include <locale.h>
+#include <cstring>
+#include <cstdio>

+using namespace std;
 using namespace log4cxx;
 using namespace log4cxx::helpers;
