From 079c01ac76baa05c7e4e5358434597551065b594 Mon Sep 17 00:00:00 2001
From: Michael Smith <michael.smith@puppetlabs.com>
Date: Wed, 31 Dec 2014 15:00:51 -0800
Subject: [PATCH] Update curl command to use latest end-point

---
 coverity-submit | 7 +++----
 1 file changed, 3 insertions(+), 4 deletions(-)

diff --git a/coverity-submit b/coverity-submit
index 90ec215..c31a45f 100755
--- a/coverity-submit
+++ b/coverity-submit
@@ -182,12 +182,11 @@ if __name__ == '__main__':
     print("Posting the analysis request...")
     do_or_die('''curl \
     --form file=@%(tmpdir)s/%(tarball)s \
-    --form project=%(covname)s \
     --form token=%(token)s \
     --form email=%(email)s \
-    --form version=%(build_version)s \
-    --form description=%(description)s \
-     http://scan5.coverity.com/cgi-bin/upload.py \
+    --form version=%(build_version)s \
+    --form description=%(description)s \
+     https://scan.coverity.com/builds?project=%(covname)s \
      ''' % globals())
     try:
         os.remove(os.path.join(tmpdir, tarball))
-- 
2.2.1

